src/candy-machine/candy-machine.service.ts

```diff
@@ -138,17 +172,17 @@ export class CandyMachineService {
     guardParams,
```

Comment:
in general I think this function is too big. It makes hard to read (for me as I read it first time).
Might be worth to make it a little bit cleaner (not saying you need to do it know, just another suggestion)

Reply:
completely agree, i'm gonna break it into 2 functions